examples/sui-move/closed_loop_token/examples/loyalty.move

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This Example implements loyalty points. The points are earned by the customer
+/// when they perform an incentivized action (e.g. purchase a product). The points
+/// are not transferable, however, they can be joined and split and can be spent
+/// on a product or service.
+module closed_loop::loyalty {
+    use std::option;
+
+    use sui::tx_context::{sender, TxContext};
+    use sui::object::{Self, UID};
+    use sui::transfer;
+
+    use closed_loop::closed_loop::{
+        Self as cl, CLPolicy, Join, Split,
+        Mint, Burn, Resolver,
+        TempToken
+    };
+
+    /// Error code for the case when the user does not have enough points.
+    const ENotEnough: u64 = 0;
+
+    /// The amount of reward for the incentivized action.
+    const REWARD_AMOUNT: u64 = 100;
+    /// Let's make it so the user needs to do 2 actions to get a "plushy bara".
+    const PLUSHY_BARA_PRICE: u64 = 200;
+
+    /// The One Time Witness for the application and the type of the closed loop
+    /// token.
+    struct LOYALTY has drop {}
+
+    // Closed Loop token allows its creator to set custom resolvers and integrate
+    // them into an application. So we need an application :)
+
+    /// The central place in the Loyalty program. Shall we rename it? Let's call
+    /// it App to make more generic and less marketplace-y.
+    struct App has key {
+        id: UID,
+        /// Special resolvers for the application. Allows minting Tokens.
+        mint_resolver: Resolver<LOYALTY, Mint>,
+        /// This resolver will help us deal with purchases.
+        burn_resolver: Resolver<LOYALTY, Burn>,
+    }
+
+    /// Freely tradable and transferable object. Can only be purchased for X
+    /// amount of loyalty points.
+    struct PlushyBara has key, store {
+        id: UID
+    }
+
+    /// In the module initializer we create a new `CLPolicy`, share it and set
+    /// the rules for interaction following the basic description of what user
+    /// can do.
+    fun init(otw: LOYALTY, ctx: &mut TxContext) {
+        let (cl_policy, cl_cap) = cl::new_token(otw, ctx);
+
+        // User can Join and Split tokens into one.
+        cl::allow<LOYALTY, Join>(&cl_cap, &mut cl_policy);
+        cl::allow<LOYALTY, Split>(&cl_cap, &mut cl_policy);
+
+        let mint_resolver = cl::create_resolver<LOYALTY, Mint>(
+            &cl_cap, &mut cl_policy, option::none(), option::none(), ctx
+        );
+
+        let burn_resolver = cl::create_resolver<LOYALTY, Burn>(
+            &cl_cap, &mut cl_policy, option::none(), option::none(), ctx
+        );
+
+        // This is our application, created and shared only once.
+        let app = App {
+            id: object::new(ctx),
+            mint_resolver,
+            burn_resolver
+        };
+
+        transfer::public_transfer(cl_cap, sender(ctx));
+        transfer::public_share_object(cl_policy);
```

Comment:
Would it make sense to put the policy inside of `App` (since it seems like they will always be used together)?

Reply:
Quite the opposite, while it's convenient for testing to return it, I'm thinking to force share it because default operations such as transfer / split are available to everyone as long as they're "allow"-ed in the Policy.

Reply:
So yes, they're used together because mint / burn require it, however, to perform a transfer you don't need App but need CLPolicy.